lib/rules/class-methods-use-this.js

```diff
@@ -45,7 +45,7 @@ module.exports = {
     },
     create(context) {
         const config = Object.assign({}, context.options[0]);
-        const exceptMethods = new Set(config.exceptMethods || []);
+        const exceptMethods = (config.exceptMethods || []).map(e => new RegExp(e, "u"));
```

Comment:
accepting a regex opens the rule up to a bunch of CVEs. is there a reason this can’t be simple globs?

Reply:
Hm okay, in my case globs would be enough. But other rules also use RegExp for exceptions camelcase e.g.

Reply:
I don't think this is a security issue if the regexes come from a config. If someone has the ability to edit a config file and add malicious regexes, they probably also have the ability to create a JS file and do worse things anyway.

If inline config comments are enabled, someone could maybe create a piece of code that takes a very long time to lint, but I think that's an accepted risk at this point.

Reply:
That’s true, but that won’t stop tons of false positive CVEs.

Reply:
With an exception or two, it seems to have stopped them so far. False-positive CVEs are sometimes a problem, but it seems silly to not use a feature for fear of false positive security reports. (Maybe someone can convince V8 to use linear-time regex matching for regexes that don't have backreferences, and then we stop tons of real CVEs.)